04.Keyframe/main.cpp

```diff
@@ -144,15 +144,17 @@ int main()
 
 		auto createSpecializedShaderFromSource = [=](const char* source, asset::ISpecializedShader::E_SHADER_STAGE stage)
 		{
-			// TODO: Update: should use the compiler set from asset manager
-			auto spirv = device->getAssetManager()->getGLSLCompiler()->createSPIRVFromGLSL(source, stage, "main", "runtimeID");
+			asset::IShaderCompiler::SCompilerOptions options;
+			options.stage = stage;
+			options.entryPoint = "main";
+			options.preprocessorOptions.sourceIdentifier = "runtimeID";
+			auto spirv = device->getAssetManager()->getCompilerSet()->compileToSPIRV(source, options);
 			return core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(spirv),asset::ICPUSpecializedShader::SInfo{ nullptr,nullptr,"main",stage });
 		};
 		// origFilepath is only relevant when you have filesystem #includes in your shader
 		auto createSpecializedShaderFromSourceWithIncludes = [&](const char* source, asset::ISpecializedShader::E_SHADER_STAGE stage, const char* origFilepath)
 		{
-			auto resolved_includes = device->getAssetManager()->getGLSLCompiler()->resolveIncludeDirectives(source, stage, origFilepath);
-			return createSpecializedShaderFromSource(reinterpret_cast<const char*>(resolved_includes->getContent()->getPointer()), stage);
+			return createSpecializedShaderFromSource(source, stage);
```

Comment:
how to replace usages of `resolveIncludeDirectives`:

A. If you remove usage of `resolveIncludeDirectives` you should fill in `SCompilerOptions::includeFinder` in order for the compile function to preprocess and resolve the includes. (get the default include finder from the appropriate compiler)

B. if you want to replace the usage of the function, it's just renamed to `preprocessShader` with a few tweaks in the CShaderCompiler function. (get the appropriate compiler based on the language first and then call the preprocessShader function)

Prefer using **A**